tests/local/helpers/roles_test.php

```diff
@@ -37,7 +37,15 @@
  * @covers \mod_bigbluebuttonbn\local\helpers\roles
  * @coversDefaultClass \mod_bigbluebuttonbn\local\helpers\roles
  */
-class roles_test extends testcase_helper {
+class roles_test extends \advanced_testcase {
+    use testcase_helper_trait;
+    /**
+     * Setup basic
+     */
+    public function setUp(): void {
```

Comment:
This change shoulnd't be necessary. no need to rename setUp to basic_setup in the trait.

Reply:
I went into a different direction: the setup was just declaring the generator, which in most case is used once, or if more can also be a local variable. I removed the setup completely. If it was more complex then we might need to use a method like you stated in your first comment. I will open another pull request.